### PR TITLE
[extract] Enhancement

### PR DESCRIPTION
**Pull request by Aloento**
`SinglePageBase` is deprecated.
`Into` can now accept `Response`.

### CODE REVIEW

**Review by outcatcher — CHANGES_REQUESTED** [review]
1. This PR contains unrelated changes
2. Changes of this PR breaking tests

**Comment by outcatcher** [comment]
Those are risky changes related to response body handling, they should not have any of existing tests failing

**Comment by Aloento** [comment]
Only map[string]interface{} of ExtractMetadata in blockstorage will be affected.
I have solved it in #400. @outcatcher

File: internal/extract/json.go [hunk 1]
```diff
@@ -122,32 +126,35 @@ func Into(body interface{}, to interface{}) error {
...
 		return fmt.Errorf("expected pointer, got %v", k)
 	}
-	switch t.Elem().Kind() {
-	case reflect.Struct:
-		return intoPtr(body, to, label)
-	default:
-		return fmt.Errorf("expected pointer to struct, got: %v", t)
+
+	if kind != t.Elem().Kind() {
+		return fmt.Errorf("expected pointer to %v, got: %v", kind, t)
```
Comment (outcatcher) [inline internal/extract/json.go]: `kind` is an int, it will be logged as an int (`25` for `Struct`)

File: pagination/single.go [hunk 2]
```diff
@@ -7,15 +7,16 @@ import (
 	"github.com/opentelekomcloud/gophertelekomcloud"
 )
 
-// SinglePageBase may be embedded in a Page that contains all of the results from an operation at once.
+// SinglePageBase may be embedded in a Page that contains all the results from an operation at once.
+// Deprecated
```
Comment (outcatcher) [inline pagination/single.go]: Deprecation message should point to new preferred way of doing this or describe why this should not be done at all.

File: internal/extract/json.go (hunk 1 above)
Comment (Aloento) [reply internal/extract/json.go]: done

File: pagination/single.go (hunk 2 above)
Comment (Aloento) [reply pagination/single.go]: done

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]
Looks good. Hope that it didn't break anything

**Comment by Aloento** [comment]
Same of https://github.com/opentelekomcloud/gophertelekomcloud/pull/403#issuecomment-1233175135

**Comment by Aloento** [comment]
regate

**Pull request merged**